Jetpack E2E: add spec for Jetpack Blaze.

worldomonation opened the pull request:
Related to https://github.com/Automattic/wp-calypso/issues/80730.

## Proposed Changes

This PR adds a new spec to test the Blaze dashboard and part of the process of Blazing a post.

Key flows:
- Tools > Advertising dashboard loading.
- entering details to Blaze a post.


## Testing Instructions

Ensure the following build configurations are passing:
  - [x] Calypso E2E (desktop)
  - [x] Jetpack E2E Simple (mobile)
  - [x] Jetpack E2E Simple (desktop)
  - [x] Jetpack E2E AT (desktop)

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-ajp-p2)?

worldomonation committed:
packages/calypso-e2e/src/lib/pages/advertising-page.ts
- new page representing Tools > Advertising.

packages/calypso-e2e/src/lib/pages/blaze-campaign-page.ts
- new page representing the Blaze Campaign.

test/e2e/specs/tools/advertising__promote.ts
- new spec to test the Blaze dashboard loading and part of the process of Blazing a post.

worldomonation committed:
test/e2e/specs/tools/advertising__promote.ts
- add test documentation.
- fix import

dpasque commented on test/e2e/specs/tools/advertising__promote.ts:
```diff
@@ -0,0 +1,85 @@
+			}
+		} );
+
+		it( 'Click on Promote for the first post', async function () {
+			await advertisingPage.clickButton( 'Promote', { row: 0 } );
+		} );
```
**Suggestion (non-blocking):** Since we need one post, do you think we should make on with the REST API for the rare case of right after the blog being emptied? WDYT?

worldomonation replied:
Hmm, that makes sense, let's do that.

worldomonation replied:
Added setup and teardown to ensure at least one post available.

dpasque reviewed (approved):
Great looking test!! 👨‍🍳

worldomonation committed:
test/e2e/specs/tools/advertising__promote.ts
- update method names
- add setup and teardown for posts.

packages/calypso-e2e/src/lib/pages/advertising-page.ts
- rename method `clickButton` to `clickButtonByNameOnRow` to better reflect its purpose.

worldomonation committed:
test/e2e/specs/tools/advertising__promote.ts
- add reference to internal P2.
